jsonrpcclient/__init__.py

```diff
@@ -2,17 +2,32 @@
 from .http_client import HTTPClient
 
 
-def request(endpoint, method, *args, **kwargs):
+def notify(endpoint, method, *args, trim_log_values=False, validate=True, **kwargs):
     """
     A convenience function. Instantiates and executes a HTTPClient request, then
     throws it away.
     """
-    return HTTPClient(endpoint).request(method, *args, **kwargs)
+    return HTTPClient(
+        endpoint, trim_log_values=trim_log_values, validate=validate
+    ).notify(method, *args, **kwargs)
 
 
-def notify(endpoint, method, *args, **kwargs):
+def request(
+    endpoint,
+    method,
+    *args,
+    id_generator=None,
+    trim_log_values=False,
+    validate=True,
+    **kwargs
+):
```

Comment:
My only concern with this approach is that now users of the library won't be able to send any of these kwargs as json-rpc parameters. I suggest allowing the first argument to be either:

```python
# pass a dict as first argument, allows "validate" as an argument to the jsonrpc call.
jsonrpcclient.request({"endpoint": "http://127.0.0.1:8080", "trim_log_values": True, "validate": True}, "myMethod", param=1, validate=False)
# with just a string, assume it's the endpoint
jsonrpcclient.request("http://127.0.0.1:8080", "myMethod", param=1, validate=False)
```

"trim_log_values" and "id_generator", seem like unlikely names for json-rpc calls, but "validate" seems generic enough that it could bite someone down the road.

If it's kept this way then I think the documentation should warn about this limitation of using the convenience methods.

Reply:
You're right.

There's also a `request_id` param which lets you set the "id" part of the request.

```python
>>> request("http://localhost:5000", "ping", request_id="foo")
--> {"jsonrpc": "2.0", "method": "ping", "id": "foo"}
```

I think it's just a consequence of using `**kwargs`, these names can't be used as JSON-RPC parts. I'll change the `validate` param to something more obscure, and make all this clear in the docs.